src/doc/unstable-book/src/language-features/proc-macro.md

```diff
@@ -6,5 +6,243 @@ The tracking issue for this feature is: [#38356]
 
 ------------------------
 
+This feature flag guards the new procedural macro features as laid out by [RFC 1566], which alongside the now-stable 
+[custom derives], provide stabilizable alternatives to the compiler plugin API (which requires the use of 
+perma-unstable internal APIs) for programmatically modifying Rust code at compile-time.
 
+The two new procedural macro kinds are:
+ 
+* Function-like or bang(!)-style procedural macros which are invoked like regular
+`macro_rules!` macros, and:
 
+* Attribute procedural macros which can be applied to any item which built-in attributes can
```

Comment:
I call these 'attribute-like' since they only look like attributes and are implemented differently - also for symmetry with 'function-like'